src/Microsoft.Diagnostics.EventFlow.Core/Implementations/FilterEvaluators/CachingPropertyExpressionEvaluator.cs

```diff
@@ -85,6 +86,16 @@ private object InterpretRhsValue(Type eventPropertyValueType)
                 }
                 else return null;
             }
+            else if (eventPropertyValueType.GetTypeInfo().IsEnum)
+            {
+                object retval = null;
+                try
+                {
+                    retval = Enum.Parse(eventPropertyValueType, this.value);
```

Comment:
Did not use TryParse since both overloads are generic methods, eg TryParse<T> and we only have a Type at resolution time.